test/distributed/test_c10d_nccl.py

```diff
@@ -3846,6 +3847,55 @@ def test_timeout_dumps(self):
             torch.cuda.synchronize()
 
 
+class NCCLTraceTestTimeoutDumpOnIdleRanks(NCCLTraceTestDumpOnTimeoutBase):
+    def _check_return_codes(self, elapsed_time):
+        # the base test infra assumes processes exit with matching return codes,
+        # but we want rank0 to abort and rank1 to exit cleanly in this test
```

Comment:
Yeah.. If it was happy with (-6, 0), I wouldn't bother doing it. Also why I had to create `NCCLTraceTestDumpOnTimeoutBase` is because if I directly inherit `NCCLTraceTestDumpOnTimeout`, when using parametrize python don't allow that. 

Let me change the comment in another PR.